src/main/host/thread_ptrace.c

```diff
@@ -600,29 +796,100 @@ SysCallCondition* threadptrace_resume(Thread* base) {
             case THREAD_PTRACE_CHILD_STATE_TRACE_ME:
                 debug("THREAD_PTRACE_CHILD_STATE_TRACE_ME");
                 break;
+            case THREAD_PTRACE_CHILD_STATE_IPC_SYSCALL: {
+                debug("THREAD_PTRACE_CHILD_STATE_IPC_SYSCALL");
+                SysCallReturn ret = syscallhandler_make_syscall(thread->sys, &thread->syscall_args);
+                switch (ret.state) {
```

Comment:
Nested switch statements? Could we break this out into its own function for readability? I would also prefer to break out the longer switch cases too when that is feasible.